### PR TITLE
T2D-230 3rd-Party-OAuth

### PR DESCRIPTION
**Pull request by selvaebi**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/ampt2d/metadata/config/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableResourceServer;
+import org.springframework.security.oauth2.config.annotation.web.configuration.ResourceServerConfigurerAdapter;
+
+public class SecurityConfig {
+
+    @ConditionalOnProperty(value = "security.enabled", havingValue = "true")
```
Comment (jorizci) [inline src/main/java/uk/ac/ebi/ampt2d/metadata/config/SecurityConfig.java]: Extract this to two classes please.

Comment (selvaebi) [reply src/main/java/uk/ac/ebi/ampt2d/metadata/config/SecurityConfig.java]: fixed in 27fa644

File: pom.xml [hunk 2]
```diff
@@ -61,17 +57,26 @@
         <dependency>
             <groupId>com.h2database</groupId>
             <artifactId>h2</artifactId>
-            <scope>test</scope>
+             <scope>test</scope>
+         </dependency>
+         <dependency>
+             <groupId>org.postgresql</groupId>
+             <artifactId>postgresql</artifactId>
+             <version>9.4-1206-jdbc42</version>
```
Comment (cyenyxe) [inline pom.xml]: This version of the driver won't work with PostgreSQL 10 (which is being adopted now). Could it be upgraded to 42.2.5?

Comment (selvaebi) [reply pom.xml]: fixed in c973edc

**Pull request merged**